### PR TITLE
Standardize testing output: Response Selection

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
- Plot histogram with confidence distribution for response selection
- Write correct and incorrect predictions for response selection

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/core/test.py [hunk 1]
```diff
@@ -604,7 +604,7 @@ def plot_story_evaluation(
     from sklearn.metrics import confusion_matrix
     from sklearn.utils.multiclass import unique_labels
     import matplotlib.pyplot as plt
-    from rasa.nlu.test import plot_confusion_matrix
```
Comment (dakshvar22) [inline rasa/core/test.py:607]: A bit unlikely that it happens but it could break someone's code relying on this function in some glue code of their own. Should we add a deprecation warning or keep a dummy function in `rasa.nlu.test` which just calls the one in `rasa.utils.plotting`?

File: tests/nlu/test_evaluation.py [hunk 2]
```diff
@@ -485,7 +495,13 @@ def test_response_evaluation_report(tmpdir_factory):
         ),
     ]
 
-    result = evaluate_response_selections(response_results, report_folder)
+    result = evaluate_response_selections(
+        response_results,
+        report_folder,
+        successes=False,
+        errors=False,
+        disable_plotting=True,
```
Comment (dakshvar22) [inline tests/nlu/test_evaluation.py:503]: Should we add a test with `disable_plotting=False` and check if the plot is actually stored?

Comment (tabergma) [reply tests/nlu/test_evaluation.py:503]: It is tested in the `cli` package.

File: rasa/core/test.py (hunk 1 above)
Comment (tabergma) [reply rasa/core/test.py:607]: Not sure if that is needed as it will be released with 2.0. I also removed two cli options for naming the histogram and confusion matrix file which are not useful anymore. If you think it is needed, I'll leave those and add a deprecation warning.

**Comment by dakshvar22** [comment]
@tabergma I pushed a small fix to add the `response_key` to the `intent` part of response selection results so that when viewing the error file, the dict looks like - 
```
{
    "text": "How do I know someone is actually doing something for my offsets?",
    "intent_target": "faq/is-this-legit",
    "response": "All offset projects have to pass a rigorous verification process set by https://goldstandard.org .",
    "response_prediction": {
      "name": "The cost per ton depends on the project you donate to. Some projects can remove a ton of CO2 for less than $1!",
      "confidence": 0.15156520903110504
    }
```
instead of 
```
{
    "text": "How do I know someone is actually doing something for my offsets?",
    "intent_target": "faq",
    "response": "All offset projects have to pass a rigorous verification process set by https://goldstandard.org .",
    "response_prediction": {
      "name": "The cost per ton depends on the project you donate to. Some projects can remove a ton of CO2 for less than $1!",
      "confidence": 0.15156520903110504
    }
```

File: rasa/core/test.py (hunk 1 above)
Comment (tabergma) [reply rasa/core/test.py:607]: @tmbo Just wanted to double check if this is fine.

**Comment by dakshvar22** [comment]
@tabergma In the confusion matrix should we use the `intent/response_key` as the labels of the rows and columns instead of the actual responses? The mapping is deterministic since there is only one response per `intent/response_key` but it would make the confusion matrix fit well when you have large number of unique responses. What do you think?

**Comment by tabergma** [comment]
@dakshvar22 Good idea. Will try it now.

**Comment by dakshvar22** [comment]
Ahh, that would need a bigger refactor. Because getting the `intent/response_key` of the predicted response is not trivial right now. Let's keep it as a separate issue.

**Comment by tabergma** [comment]
😄 Ok. As an alternative, we could also maybe use a legend that maps the actual response to `A`, `B`, etc. What do you think?

**Comment by dakshvar22** [comment]
I would say let us directly replace them with the `intent/response_key` separately because that's the most ideal.
